### PR TITLE
Package spark

### PR DESCRIPTION
**Pull request by mserranom**
#### Pull Request Checklist:
- [x] Tests for the changes have been added
- [x] Docs have been added / updated

### What kind of change does this PR introduce?

A new `blurr package-spark --source-dir my-app` has been added. Check [`__main.py__`](https://github.com/productml/blurr/pull/288/files#diff-45b523a96425ea20f4f6ca353fb24bbf) for docs

#### Link to issue
fixes #288

### CODE REVIEW

File: blurr/__main__.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Usage:
     blurr validate [--debug] [<DTC> ...]
     blurr transform [--debug] [--runner=<runner>] [--streaming-dtc=<dtc-file>] [--window-dtc=<dtc-file>] [--data-processor=<data-processor>] (--source=<raw-json-files> | <raw-json-files>)
+    blurr package-spark [--debug] [--source-dir=<dir>] [--target=<zip-file>]
```
Comment (indradeep) [inline blurr/__main__.py]: Why not call the command `package` and take `spark` as a parameter.  

e.g.  the `blurr package [--runner spark][--debug] [--source-dir=<dir>] [--target=<zip-file>]`?

File: tests/cli/package_spark_test.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    return cli(arguments)
+
+
+def generate_target_filename():
+    return os.path.join(TMP, str(uuid.uuid4()) + '.zip')
+
+
+def get_running_validation_str(file_name: str) -> str:
+    return 'Running syntax validation on tests/core/syntax/dtcs/' + file_name
```
Comment (indradeep) [inline tests/cli/package_spark_test.py]: Text Needs updating?

Comment (mserranom) [reply tests/cli/package_spark_test.py]: ooops, this method shouldn't be there, deleted.

File: blurr/__main__.py (hunk 1 above)
Comment (mserranom) [reply blurr/__main__.py]: `package` would be an unnecessary abstraction, there's only spark packaging. If we end up supporting something else, let's change.

File: blurr/__main__.py [hunk 3]
```diff
@@ -17,6 +18,8 @@
                     3. Both streaming and window DTC are provided - Transform
                     outputs the final result of applying the streaming and window
                     DTC on the raw data file.
+    package-spark   Generates a submittable Spark app with .zip extension.
```
[the remark this answers is not in the document]
Comment (mserranom) [reply blurr/__main__.py]: added extra docs

File: blurr/cli/package_spark.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+from distutils.dir_util import copy_tree
+
+import os
+from pip._internal import main as pip_main
+
+
+import tempfile
+
+from blurr.cli.util import eprint
+from blurr.core import logging
```
[the remark this answers is not in the document]
Comment (mserranom) [reply blurr/cli/package_spark.py]: Done

File: blurr/cli/package_spark.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def _create_zip(dir: str, target_file: str) -> None:
+
```
[the remark this answers is not in the document]
Comment (mserranom) [reply blurr/cli/package_spark.py]: deleted

**Pull request merged**